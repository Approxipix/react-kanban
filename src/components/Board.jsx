import React, { Component, PureComponent } from 'react';
import Column from './Column'
import styled from 'styled-components';
import { DragDropContext, Droppable } from 'react-beautiful-dnd';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import ColumnAdder from './ColumnAdder';
import { moveList, moveTask, deleteBoard } from "../redux/rootReducer/actions";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { history } from '../redux/store';

const BoardWrapper = styled.div`
  height: 100%;
  padding: 1rem 1.5rem;
  background-color: #2E7EAF;
`;

const Container = styled.div`
  display: flex;
  align-items: flex-start;
`;

const BoardHeader = styled.div`
  margin-bottom: 1rem;
  padding: 0 .5rem;
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

const Actions = styled.div`

`;

const Button = styled.button`
  padding: .5rem;
  cursor: pointer;
  color: #fff;
  border-radius: .2rem;
  transition: background .2s ease-in-out;
  &:hover {
    background: rgba(0, 0, 0, 0.2);
  }
`;

const Title = styled.h1`
  margin: 0;
  color: #fff;
  font-size: 1.3rem;
  font-weight: 600;
`;


class InnerList extends Component {
  render() {
    return (
      <>
        {this.props.list.map((list, index) => (
          <Column
            boardIndex={this.props.boardIndex}
            listIndex={index}

            key={list.listId}
            column={list}
            index={index}
          />
        ))}
        <ColumnAdder boardIndex={this.props.boardIndex}/>
      </>
    )
  }
}

class Board extends Component {
  onDragEnd = (result) => {
    const { destination, source, draggableId, type } = result;
    if (!destination) return;
    if (
      destination.droppableId === source.droppableId &&
      destination.index === source.index
    ) return;

    if (type === 'column') {
      let boardIndex;
      this.props.boards.forEach((board, index) => {
        if (board.boardId === +this.props.match.params.id) {
          boardIndex = index;
        }
      });
      this.props.actions.moveList({
        sourceIndex: source.index,
        destinationIndex: destination.index,
        index: boardIndex,
      });
    }

    if (type === 'task') {
      let boardIndex;
      this.props.boards.forEach((board, index) => {
        if (board.boardId === +this.props.match.params.id) {
          boardIndex = index;
        }
      });
      this.props.actions.moveTask({
        sourceeIndex: source.index,
        sourceListIndex: source.droppableId,
        destinationnIndex: destination.index,
        destinationListIndex: destination.droppableId,
        indeex: boardIndex,
      });
    }
  };

  render() {
    const { boards } = this.props;
    let boardIndex;
    const board = boards.find((board, index) => {
      boardIndex = index;
      return board.boardId === +this.props.match.params.id
    });
    return (
      <BoardWrapper>
        <BoardHeader>
          <Title>
            {board.title}
          </Title>
          <Actions>
            <Button onClick={() => {
              this.props.actions.deleteBoard(+this.props.match.params.id);
              history.push("/boards")
            }}>
              <FontAwesomeIcon icon="trash" />
              Delete board
            </Button>
          </Actions>
        </BoardHeader>
        <DragDropContext
          onDragEnd={this.onDragEnd}
        >
          <Droppable
            droppableId='all-columns'
            direction='horizontal'
            type='column'
          >
            {(provided) => (
              <Container
                {...provided.droppableProps}
                ref={provided.innerRef}
              >
                <InnerList
                  boardIndex={boardIndex}
                  list={board.list}
                />
                {provided.placeholder}
              </Container>
            )}
          </Droppable>
        </DragDropContext>
      </BoardWrapper>
    )
  }
}

function mapStateToProps(state) {
  return {
    boards: state.rootReducer.boards,
  }
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      moveList: moveList,
      moveTask: moveTask,
      deleteBoard: deleteBoard,
    }, dispatch)
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Board);
