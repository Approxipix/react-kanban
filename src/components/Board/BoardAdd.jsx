import React, { Component } from 'react';
import { connect } from 'react-redux';
import { history } from "../../redux/store";
import { bindActionCreators } from 'redux';
import { addBoard } from '../../redux/boardReducer/actions/actions';
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { Actions, Input, SubmitButton, CancelButton } from '../BaseComponent';
import ClickOutside from "../ClickOutside/ClickOutside";
import styled from 'styled-components';
import PropTypes from "prop-types";

const AddButton = styled.button`
  font-size: 1rem; 
  color: #172b4d;
  font-weight: 600;
  background-color: #dadde3;
  border-radius: .4rem
  transition: background-color .2s ease-in;
  cursor: pointer;
  &:hover {
    background-color: #c8cace;
  }
`;
AddButton.displayName = 'AddButton';

const Container = styled.div`
  height: 100%;
  padding: 1rem .6rem;
  background-color: #dadde3;
  border-radius: .4rem
`;

const ColorPicker = styled.div`
  display: flex;
  padding: .25rem 0;
  margin-bottom: .5rem;
`;
ColorPicker.displayName = 'ColorPicker';

const ColorPickerItem = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 1.5rem;
  height: 1.5rem;
  font-size: .7rem;
  color: #fff;
  background-color: ${props => props.color};
  border-radius: .4rem;
  cursor: pointer;
  &:not(:last-child) {
    margin-right: .5rem;
  }
  &:hover {
    box-shadow: inset 0 0 0 10rem rgba(0, 0, 0, .1)
  }
`;

class BoardAdd extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isOpened: false,
      boardTitle: '',
      boardColor: props.colors[0] || '',
    };
  }

  handleKeyDown = (e) => {
    if (e.keyCode === 27) {
      this.toggleOpened();
    }
  };

  handleChange = (key, value) => {
    this.setState({
      ...this.state,
      [key]: value,
    });
  };

  toggleOpened = () => {
    this.setState({
      isOpened: !this.state.isOpened,
      boardTitle: '',
    })
  };

  handleSubmit = (e) => {
    e.preventDefault();
    const { boardTitle } = this.state;
    if (!boardTitle) return;
    const boardId = Date.now().toString();
    this.props.actions.addBoard({
      boardTitle: boardTitle,
      boardColor: this.state.boardColor,
      boardId: boardId,
    });
    history.push(`/b/${boardId}`)
  };

  render = () => {
    const { colors } = this.props;
    const { isOpened, boardTitle, boardColor } = this.state;
    return isOpened ? (
      <ClickOutside toggleOpened={this.toggleOpened}>
        <Container color={boardColor}>
          <form onSubmit={this.handleSubmit}>
            <Input
              type="text"
              placeholder="Add board title"
              value={boardTitle}
              onKeyDown={this.handleKeyDown}
              onChange={(e) => this.handleChange('boardTitle', e.target.value)}
              spellCheck={false}
              autoFocus
            />
            <ColorPicker>
              {colors.map((color, index) => (
                <ColorPickerItem
                  key={index}
                  color={color}
                  onClick={() => this.handleChange('boardColor', color)}
                >
                  {boardColor === color && <FontAwesomeIcon icon="check"/>}
                </ColorPickerItem>
              ))}
            </ColorPicker>
            <Actions>
              <SubmitButton type='submit' disabled={!boardTitle}>
                Create board
              </SubmitButton>
              <CancelButton onClick={() => this.toggleOpened()}>
                <FontAwesomeIcon icon="times"/>
              </CancelButton>
            </Actions>
          </form>
        </Container>
      </ClickOutside>
    ) : (
      <AddButton onClick={() => this.toggleOpened()}>
        Create new board...
      </AddButton>
    );
  };
}

BoardAdd.defaultProps = {
  colors: [],
};

BoardAdd.propTypes = {
  colors: PropTypes.arrayOf(PropTypes.string).isRequired,
};

function mapStateToProps(state) {
  return {
    colors: state.rootReducer.colors,
  }
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      addBoard: addBoard,
    }, dispatch)
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(BoardAdd);
