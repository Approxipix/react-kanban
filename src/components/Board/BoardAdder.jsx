import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { addBoard } from '../../redux/boardReducer/actions';
import styled from 'styled-components'
import { Input, SubmitButton, CancelButton } from '../BaseComponent';
import ClickOutside from '../ClickOutside'

const AddButton = styled.button`
  font-size: 1rem; 
  color: #444;
  font-weight: 600;
  background-color: #ddd;
  border-radius: .4rem
  transition: background-color .1s;
  cursor: pointer;
  &:hover {
    background-color: #ccc;
  }
`;

const Wrapper = styled.div`
  height: 100%;
  padding: .6rem;
  background-color: #ddd;
  border-radius: .4rem
`;

const Title = styled.h2`
  margin-bottom: .5rem;
  font-size: 1rem;
`;

const Actions = styled.div`
  display: flex;
  align-items: center;
`;

class BoardAdder extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isOpened: false,
      boardTitle: '',
    };
  }

  handleChange = (e) => {
    this.setState({
      boardTitle: e.target.value
    });
  };

  handleKeyDown = (e) => {
    if (e.keyCode === 27) {
      this.setState({
        isOpened: false
      });
    }
  };

  toggleOpened = () => {
    this.setState({
      isOpened: !this.state.isOpened,
      boardTitle: '',
    })
  };

  handleSubmit = (e) => {
    e.preventDefault();
    const { boardTitle } = this.state;
    if (!boardTitle) return;
    this.props.actions.addBoard({
      boardTitle: boardTitle,
      boardColor: this.props.colors[0]
    });
    this.setState({
      boardTitle: '',
    });
    this.textInput.focus();
  };

  render = () => {
    const { isOpened, boardTitle } = this.state;
    return isOpened ? (
      <ClickOutside toggleOpened={this.toggleOpened}>
        <Wrapper>
          <Title>
            New Board
          </Title>
          <form id="board-add-form" onSubmit={this.handleSubmit}>
            <Input
              autoFocus
              ref={(input) => { this.textInput = input; }}
              type="text"
              placeholder="Add board title"
              value={boardTitle}
              onKeyDown={this.handleKeyDown}
              onChange={this.handleChange}
              spellCheck={false}
            />
            <Actions>
              <SubmitButton type="submit" disabled={boardTitle === ""}>
                Create board
              </SubmitButton>
              or
              <CancelButton onClick={() => this.toggleOpened()}>
                cancel
              </CancelButton>
            </Actions>
          </form>
        </Wrapper>
      </ClickOutside>
    ) : (
      <AddButton onClick={() => this.toggleOpened()}>
        Create new board...
      </AddButton>
    );
  };
}

function mapStateToProps(state, ownProps) {
  return {
    colors: state.rootReducer.colors,
  }
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      addBoard: addBoard,
    }, dispatch)
  };
}


export default connect(mapStateToProps, mapDispatchToProps)(BoardAdder);
