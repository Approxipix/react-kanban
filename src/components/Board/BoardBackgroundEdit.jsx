import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { editBoardColor } from '../../redux/boardReducer/actions/actions';
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import styled from 'styled-components'

const Container = styled.div`
  position: relative;
`;

const List = styled.ul`
  position: absolute;
  top: 90%;
  display: grid;
  grid-template-columns: repeat(2, 2fr);
  grid-gap: .75rem;
  width: 100%;
`;

const Item = styled.li`
`;

const Button = styled.button`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 5rem;
  background-color: ${props => props.value};
  border-radius: .5rem;
  color: #fff;
  cursor: pointer;
  &:hover {
    box-shadow: inset 0 0 0 10rem rgba(0, 0, 0, .1)
  }
`;

class BoardBackgroundEdit extends Component {
  editBoardColor = (color) => {
    this.props.actions.editBoardColor({
      boardId: this.props.boardId,
      boardColor: color
    });
  };

  render() {
    const { boardColor, colors } = this.props;
    return (
      <Container>
        <List>
          {colors.map((color, index) => (
            <Item key={index}>
              <Button value={color} onClick={() => this.editBoardColor(color)}>
                {boardColor === color && <FontAwesomeIcon icon="check" />}
              </Button>
            </Item>
          ))}
        </List>
      </Container>
    );
  }
}

function mapStateToProps(state) {
  return {
    colors: state.rootReducer.colors,
  }
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      editBoardColor: editBoardColor,
    }, dispatch)
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(BoardBackgroundEdit);
