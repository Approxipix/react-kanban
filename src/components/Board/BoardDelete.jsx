import React, { Component } from 'react';
import { connect } from 'react-redux';
import { history } from "../../redux/store";
import { bindActionCreators } from 'redux';
import { deleteBoard } from '../../redux/boardReducer/actions/actions';
import { Text, SubmitButton } from '../BaseComponent';

class BoardDelete extends Component {
  deleteBoard = () => {
    this.props.actions.deleteBoard({
      boardId: this.props.boardId
    });
    history.push('/boards')
  };

  render() {
    return (
      <>
        <Text>
          All actions will be removed from the activity feed and you wonâ€™t be able to re-open the board.
          There is no undo.
        </Text>
        <SubmitButton color={'danger'} onClick={() => this.deleteBoard()}>
          Delete board
        </SubmitButton>
      </>
    );
  }
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      deleteBoard: deleteBoard,
    }, dispatch)
  };
}

export default connect(null, mapDispatchToProps)(BoardDelete);
