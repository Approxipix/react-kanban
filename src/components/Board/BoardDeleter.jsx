import React, { Component } from 'react';
import { connect } from "react-redux";
import { bindActionCreators } from 'redux';
import { deleteBoard } from "../../redux/boardReducer/actions";
import { history } from "../../redux/store";
import styled from "styled-components";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";


const Button = styled.button`
  margin: 0.5rem;
  padding: .5rem;
  cursor: pointer;
  color: #fff;
  border-radius: .2rem;
  transition: background .2s ease-in-out;
  white-space: nowrap;
  &:hover {
    background: rgba(0, 0, 0, 0.2);
  }
`;

class BoardDeleter extends Component {
  deleteBoard = () => {
    this.props.actions.deleteBoard({
      boardId: this.props.boardId
    });
    history.push('/boards')
  };

  render() {
    return (
      <Button onClick={() => this.deleteBoard()}>
        <FontAwesomeIcon icon="trash" />
        &nbsp;Delete board
      </Button>
    )
  }
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      deleteBoard: deleteBoard,
    }, dispatch)
  };
}

export default connect(null, mapDispatchToProps)(BoardDeleter);

