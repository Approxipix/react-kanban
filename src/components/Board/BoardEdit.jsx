import React, { Component } from "react";
import { connect } from "react-redux";
import styled from 'styled-components'
import { bindActionCreators } from 'redux';
import { editBoard } from '../../redux/boardReducer/actions';

const Form = styled.form`
  width: 90%;
`;

const Input = styled.input`
  padding: .3rem .5rem;
  width: 100%;
  border: none;
  font-size: 1.3rem;
  font-weight: 600;
  box-shadow: none;
  color: #40424b;
`;

class BoardEdit extends Component {
  constructor(props) {
    super(props);
    this.state = {
      title: '',
    };
  }

  componentDidMount() {
    this.setState({
      title: this.props.board.title,
    });
    document.addEventListener('click', this.outerClick);
  }

  componentWillUnmount() {
    document.removeEventListener('click', this.outerClick);
  }

  outerClick = (e) => {
    let { target } = e;
    if (target.id === 'board-edit-form') {
      return;
    }
    if (!!target.closest && (target.closest('#board-edit-form'))) {
      return;
    }
    this.props.closeEdit();
  };

  handleChange = (key, value) => {
    this.setState({
      ...this.state,
      [key]: value,
    });
  };

  handleSubmit = event => {
    event.preventDefault();
    const { title } = this.state;
    if (!title)  return;
    const data = {
      boardIndex: this.props.boardIndex,
      boardTitle: title,
    };
    this.props.actions.editBoard(data);
    this.props.closeEdit()
  };

  render = () => {
    const { title } = this.state;
    return (
      <Form onSubmit={this.handleSubmit} id="board-edit-form">
        <Input
          autoFocus
          type="text"
          placeholder="Board name"
          value={title}
          onChange={(e) => this.handleChange('title', e.target.value)}
          spellCheck={false}
        />
      </Form>
    );
  };
}

function mapStateToProps(state, ownProps) {
  return {
    board: state.rootReducer.boards[ownProps.boardIndex],
  }
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      editBoard: editBoard,
    }, dispatch)
  };
}


export default connect(mapStateToProps, mapDispatchToProps)(BoardEdit);
