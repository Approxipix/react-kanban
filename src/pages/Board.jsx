import React, { Component } from 'react';
import PropTypes from "prop-types";
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { setCurrentBoardID } from "../redux/rootReducer/actions/actions";
import { moveList } from "../redux/boardReducer/actions/actions";
import { moveCard } from "../redux/listReducer/actions/actions";
import { DragDropContext, Droppable } from 'react-beautiful-dnd';
import { Redirect } from "react-router";
import BoardHeader from '../components/Board/BoardHeader';
import Sidebar from '../components/Sidebar/Sidebar';
import ListAdd from '../components/List/ListAdd';
import List from '../components/List/List'
import styled from 'styled-components';

const Wrapper = styled.div`
  position: relative;
  height: 100%;
  padding-top: .5rem;
  background-color: ${props => props.color};
  transition: background-color .2s ease-in;
`;

const Container = styled.div`
  display: flex;
  align-items: flex-start;
  flex-shrink: 0;
  height: 92%;
  max-width: 100vw;
  padding: 0 1rem 1rem;
  overflow: auto;
`;

const InnerList = (props) => {
  const { boardId, listsId, } = props;
  return (
    listsId.map((listId, index) => (
      <List
        key={index}
        index={index}
        boardId={boardId}
        listId={listId}
      />
    ))
  )
};

class Board extends Component {
  componentDidMount() {
    const boardId = this.props.match.params.boardId;
    this.props.actions.setCurrentBoardID(boardId);
  }

  componentWillUnmount() {
    this.props.actions.setCurrentBoardID(null)
  }

  handleDragEnd = (result) => {
    const { destination, source, type } = result;
    if (!destination) return;
    if (
      destination.droppableId === source.droppableId &&
      destination.index === source.index
    ) return;

    if (type === 'column') {
      this.props.actions.moveList({
        boardId: this.props.board._boardId,
        sourceIndex: source.index,
        destinationIndex: destination.index,
      });
    }

    if (type === 'task') {
      this.props.actions.moveCard({
        sourceIndex: source.index,
        sourceListIndex: source.droppableId,
        destinationIndex: destination.index,
        destinationListIndex: destination.droppableId,
      });
    }
  };

  render() {
    const { board, currentBoardId } = this.props;
    if (!currentBoardId) return null;
    if (!board) return <Redirect to='/'/>;
    return (
      <Wrapper color={board.color}>
        <BoardHeader boardId={board._boardId}/>
        <DragDropContext onDragEnd={this.handleDragEnd}>
          <Droppable
            type='column'
            droppableId={board._boardId}
            direction='horizontal'
          >
            {(provided) => (
              <Container
                {...provided.droppableProps}
                ref={provided.innerRef}
              >
                <InnerList boardId={board._boardId} listsId={board.lists}/>
                {provided.placeholder}
                <ListAdd boardId={board._boardId}/>
              </Container>
            )}
          </Droppable>
        </DragDropContext>
        <Sidebar />
      </Wrapper>
    )
  }
}

Board.defaultProps = {
  isSidebarOpened: false,
  currentBoardId: null,
  board: {},
};

Board.propTypes = {
  isSidebarOpened: PropTypes.bool.isRequired,
  currentBoardId: PropTypes.string.isRequired,
  board: PropTypes.shape({
    _boardId: PropTypes.string.isRequired,
    title: PropTypes.string,
    color: PropTypes.string,
    lists: PropTypes.arrayOf(PropTypes.string),
  }).isRequired,
};

function mapStateToProps(state) {
  return {
    isSidebarOpened: state.rootReducer.isSidebarOpened,
    currentBoardId: state.rootReducer.currentBoardID,
    board: state.boardReducer[state.rootReducer.currentBoardID],
  }
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      setCurrentBoardID: setCurrentBoardID,
      moveList: moveList,
      moveCard: moveCard,
    }, dispatch)
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Board);
