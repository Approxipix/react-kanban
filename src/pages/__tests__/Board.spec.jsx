import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import toJson from "enzyme-to-json";
import Board, { InnerList } from '../Board';

const mockStore = configureStore([]);
jest.mock('react-beautiful-dnd', () => ({
  Droppable: ({ children }) => children({
    draggableProps: {
      style: {},
    },
    innerRef: jest.fn(),
  }, {}),
  Draggable: ({ children }) => children({
    draggableProps: {
      style: {},
    },
    innerRef: jest.fn(),
  }, {}),
  DragDropContext: ({ children }) => children,
}));

describe('<Board>', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      rootReducer: {
        currentBoardID: 'BoardID',
      },
      boardReducer: {
        BoardID: {
          _boardId: 'BoardID',
          title: 'Board Title',
          color: '#b1b1b1',
          lists: [],
        },
      }
    });
    store.dispatch = jest.fn();

    const wrapper = shallow(<Board store={store} />, { disableLifecycleMethods: true });
    component = wrapper.find('Board').dive();
  });

  it('should render with given state from Redux store', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should dispatch an action when drag type is column', () => {
    const expectedAction = {
      type: 'MOVE_LIST',
      payload: {
        boardId: 'BoardID',
        sourceIndex: 0,
        destinationIndex: 1,
      }
    };

    component.instance().handleDragEnd({
      type: 'column',
      source: {
        index: 0,
        droppableId: 0,
      },
      destination: {
        index: 1,
        droppableId: 1,
      },
    });

    expect(store.dispatch).toHaveBeenCalledTimes(1);
    expect(store.dispatch).toHaveBeenCalledWith(expectedAction);
  });

  it('should dispatch an action when drag type is task', () => {
    const expectedAction = {
      type: 'MOVE_CARD',
      payload: {
        sourceIndex: 0,
        sourceListIndex: 0,
        destinationIndex: 1,
        destinationListIndex: 1,
      }
    };

    component.instance().handleDragEnd({
      type: 'task',
      source: {
        index: 0,
        droppableId: 0,
      },
      destination: {
        index: 1,
        droppableId: 1,
      },
    });

    expect(store.dispatch).toHaveBeenCalledTimes(1);
    expect(store.dispatch).toHaveBeenCalledWith(expectedAction);
  });

  describe('<InnerList>', () => {
    let component;

    beforeEach(() => {
      const props = {
        boardId: 'BoardID',
        listsId: ['List1ID', 'List2ID', 'List3ID'],
      };

      component = shallow(<InnerList {...props} />);
    });

    it('should render correctly', () => {
      expect(toJson(component)).toMatchSnapshot();
    });

    it('should render a List component for each list id', () => {
      expect(component.length).toEqual(3)
    });
  })
});
