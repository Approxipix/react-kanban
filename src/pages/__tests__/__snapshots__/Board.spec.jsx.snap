// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Board> <InnerList> should render correctly 1`] = `
Array [
  <Connect(List)
    boardId="BoardID"
    index={0}
    key="0"
    listId="List1ID"
  />,
  <Connect(List)
    boardId="BoardID"
    index={1}
    key="1"
    listId="List2ID"
  />,
  <Connect(List)
    boardId="BoardID"
    index={2}
    key="2"
    listId="List3ID"
  />,
]
`;

exports[`<Board> should render with given state from Redux store 1`] = `
<styled.div
  color="#b1b1b1"
>
  <Connect(BoardHeader)
    boardId="BoardID"
  />
  <DragDropContext
    onDragEnd={[Function]}
  >
    <Droppable
      direction="horizontal"
      droppableId="BoardID"
      type="column"
    >
      <Component />
    </Droppable>
  </DragDropContext>
  <Connect(Sidebar) />
</styled.div>
`;
